src/coreclr/vm/excep.cpp

```diff
@@ -8168,15 +8168,12 @@ void UnwindAndContinueRethrowHelperInsideCatch(Frame* pEntryFrame, Exception* pE
     pThread->SetFrame(pEntryFrame);
 
 #ifdef _DEBUG
-    if (!NingenEnabled())
-    {
-        CONTRACT_VIOLATION(ThrowsViolation);
+    CONTRACT_VIOLATION(ThrowsViolation);
```

Comment:
The method ends so we are good, but I would keep the scope here as it is subtly changed now.